### DISCUSSION TITLE
Interested in PR to disable turnouts / sensors / outputs? (save flash...)

### DISCUSSION THREAD

**Discussion by newHeiko** [DCC-EX/CommandStation-EX#374] [Ideas]
Hello everyone,

First of all, thanks for the great project that is CommandStation-EX. I've loaded the 5.0.0 release onto an Arduino Nano with a "Nano L298p" standard motor shield to form a compact programmer / test running device, similar in usage to a SPROG. Wonderful.

To add an LCD, I had to save flash space, so I went axe-in-the woods and renamed files and commented out references to those files that I didn't need - anything turnouts, sensors and outputs.

Would you be interested in a PR to add #defines "DISABLE_SENSORS", "DISABLE_TURNOUTS" and "DISABLE_OUTPUTS" to config-example.h? If yes, I'll give cleaning up my mess a shot over the holidays, but if you can say right away that won't be accepted I don't want to waste my time on it either :)

Thanks again,
Heiko

**Comment by peteGSX** [comment]
Heya @newHeiko just coming back to you on this one. If you ensure you have no "myAutomation.h" file, then this should already effectively do this for you.

**Comment by newHeiko** [comment]
Thanks for getting back to me, but no, at least for 5.0.0 on Arduino 1.8.19, it doesn't. I still get 
> Sketch uses 31660 bytes (103%) of program storage space. Maximum is 30720 bytes.

I also just tried 5.0.7 on Arduino 1.8.19. Saves two bytes:
> Sketch uses 31658 bytes (103%) of program storage space. Maximum is 30720 bytes.

That's without myAutomation.h (not included in the https://github.com/DCC-EX/CommandStation-EX/releases/download/v5.0.0-Prod/CommandStation-EX-5.0.0-Prod.tar.gz anyway) and with the following config.h changes over the example:

```
diff config.h config.example.h 
87c87
< #define ENABLE_WIFI false
---
> #define ENABLE_WIFI true
138c138
< #define ENABLE_ETHERNET false
---
> //#define ENABLE_ETHERNET true
157c157
< #define LCD_DRIVER  0x27,16,2
---
> // #define LCD_DRIVER  0x27,16,2
181c181
< #define DISABLE_EEPROM
---
> // #define DISABLE_EEPROM
```

I may well be doing something wrong (too large bootloader, maybe?) but I did get it to work by removing a couple of files and functions and commenting out the references to them (for example I removed DCCEXParser::parseT and DCCEXParser::parseS).

It works for me, but it's a mess. If you want my changes, I can clean them up, but if you can say right away you're not interested, I'll save myself the time.

Heiko

**Comment by habazut** [comment]
If you are not happy with 5.0.x (Prod), test then 5.2.x (Devel) series.
If you need a LCD_DRIVER  you need to disable more stuff. Have a look at the config.example.h for what more disable options there are.
(EEPROM, HAL, DIAG, PROG if I recall correct)

Testing is already now a mess. The Uno is a dead end if you want more than simple functionality.

Harald.

**Reply by newHeiko** [reply]
I may not be making myself clear:

I have modified DCC-EX 5.0.0 to get it working with LCD_DRIVER on a Nano. [edit]And I'm very happy with it.[/edit]

I was wondering if you were interested in my patches.

I take your statement to mean "No". That's OK, no hard feelings. It also means I can save myself the time to make them "pretty" :)

Heiko
